lib/imagefs.coffee

```diff
@@ -280,3 +327,13 @@ exports.listDirectory = (definition) ->
 			listDirectory(disk, definition.partition, definition.path)
 	else if definition.image instanceof filedisk.Disk
 		listDirectory(definition.image, definition.partition, definition.path)
+
+###*
+# @summary Closes the allocated resources. Call this when you're done using imagefs if you expect the program to end.
+# @function
+# @public
+#
+# @returns {Promise}
+###
+exports.close = ->
+	ext2fs.close()
```

Comment:
> I don't get the problem with having the ext2fs a module global.
> Would it be better if I prevented close from being called twice?

Example:

```js
// Kick off operations on several images in parallel
async.map([
  'path/to/image-1.img',
  'path/to/image-2.img',
  // ...
], function iter(filename, callback) {

  var options = {
    image: filename,
    partition: { primary: 2 },
    path: '/',
  }

  // This will not have completed by the time
  // the next `.listDirectory()` call kicks off, whichever
  // one is faster will call `extfs.close()` when disposed
  // and will shredder still ongoing operations on other images
  imagefs.listDirectory(options)
    .catch(callback)
    .then((result) => { callback(null, result) })

}, function done(error, results) {
  // ...
})
```

From what I can tell from the `extfs` docs, `extfs.close()` should only be called before the process exits and the disposer should only call `ext2fs.umount(filesystem, callback)` instead.

For `extfs.close()`, you might want to use

```js
process.on('beforeExit', function() {
  extfs.close()
})
```